lua/lspconfig/server_configurations/elixirls.lua

```diff
@@ -4,7 +4,15 @@ return {
   default_config = {
     filetypes = { 'elixir', 'eelixir', 'heex', 'surface' },
     root_dir = function(fname)
-      return util.root_pattern 'mix.exs'(fname) or util.find_git_ancestor(fname) or vim.loop.os_homedir()
+      if not fname or fname == "" then
```

Comment:
I think you don't need this check at all since we can trust there's a fname coming in.